### PR TITLE
[Impeller] Allow skipping of inactive uniforms in GLES

### PR DESCRIPTION
**Pull request by bdero**
Resolves https://github.com/flutter/flutter/issues/109966.

It turns out that "active" uniforms are defined as uniforms that contribute to the shader's output. So many (most?) drivers eliminate unused uniforms while compiling shaders anyhow. There doesn't seem to be a way to detect "inactive" uniforms that were eliminated, which means that making ImpellerC not eliminate the uniforms won't get us any closer to properly sanity checking the bindings (while also making the driver do slightly more work).

This lack of predictability in the final uniform layout is one of the reasons looking up the uniform locations is necessary in the first place.

### CODE REVIEW

**Review by jonahwilliams — APPROVED** [review]
LGTM!

File: impeller/renderer/backend/gles/buffer_bindings_gles.cc [hunk 1]
```diff
@@ -215,8 +218,9 @@ bool BufferBindingsGLES::BindUniformBuffer(const ProcTableGLES& gl,
                                                     member.array_elements > 1);
     const auto location = uniform_locations_.find(member_key);
     if (location == uniform_locations_.end()) {
-      VALIDATION_LOG << "Location for uniform member not known: " << member_key;
-      return false;
+      FML_DLOG(WARNING) << "Location for uniform member not known: "
```
Comment (chinmaygarde) [inline impeller/renderer/backend/gles/buffer_bindings_gles.cc:221]: In that case, let's get rid of this warning as well. Not that we can do anything about it if we see it.

**Comment by bdero** [comment]
Hmm, I'm still having trouble running anything through the GL backend on MacOS for some reason.

I'd appreciate if someone with working GL tests on Mac can try running `--gtest_filter="Play/RendererTest.InactiveUniforms/*"` to verify that the shader outputs green on both backends and that the warning prints on the GLES backend.

**Review by chinmaygarde — APPROVED** [review]

File: impeller/renderer/backend/gles/buffer_bindings_gles.cc (hunk 1 above)
Comment (bdero) [reply impeller/renderer/backend/gles/buffer_bindings_gles.cc]: Done.

**Comment by bdero** [comment]
Checked this on Windows.

**Pull request merged**